components/world-3/construction/buildings.tsx

```diff
@@ -24,12 +26,18 @@ export function BuildingsDisplay() {
             return [];
         }
 
+        let buildingsData = constructionData.buildings;
+
         if (hideMaxed) {
-            return constructionData.buildings.filter(building => building.level != building.maxLvl);
+            buildingsData = buildingsData.filter(building => !building.maxed);
+        }
+
+        if (onlyCurrentlyBuilding) {
```

Comment:
We don't have to change it now I guess, but I might do a future PR to make the two checkboxes exclusive to each other.

It's a very minor thing but I don't think it makes sense to have both ticked at once from a logical perspective. But it doesn't break anything right now so I'm not going to change it before merging.